src/api/extension/request.ts

```diff
@@ -54,6 +62,67 @@ const accounts = async (): Promise<string[] | Errors.Error> => {
     return [account]
 }
 
+const getAssetCoins = async (params: AssetCoinsParams) => {
+    const puzzle = await Secure.getPuzzle()
+
+    let puzzleHash = puzzle.hashHex()
+    if (params.assetId) {
+        if (params.type === AssetCoinsTypeEnum.CAT) {
+            const walletPublicKey = await Secure.getWalletPublicKey()
+            const wallet = new Wallet(walletPublicKey, {
+                hardened: true,
+                index: 0,
+            })
+            const assetId = fromHex(params?.assetId)
+            const cat = new CAT(assetId, wallet)
+            puzzleHash = cat.hashHex()
+        }
+
+        if (params.type === AssetCoinsTypeEnum.NFT) {
+            // TODO: develop when NFT implement
+        }
+
+        if (params.type === AssetCoinsTypeEnum.DID) {
+            // TODO:develop when DID implement
+        }
+    }
+
+    const spendableCoins = (
+        await getSpendableCoins({
+            puzzle_hash: puzzleHash,
+        })
+    ).data?.data
```

Comment:
need default value to prevent undefined?
```ts
const spendableCoins = (
        await getSpendableCoins({
            puzzle_hash: puzzleHash,
        })
    ).data?.data ?? []
```

Reply:
I think the default value is better!